specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2020-01-01/SecurityInsights.json

```diff
@@ -664,364 +939,1104 @@
         }
       },
       "type": "object"
     },
-    "ActionRequest": {
+    "Bookmark": {
       "allOf": [
         {
           "$ref": "#/definitions/ResourceWithEtag"
         }
       ],
-      "description": "Action for alert rule.",
+      "description": "Represents a bookmark in Azure Security Insights.",
       "properties": {
         "properties": {
-          "$ref": "#/definitions/ActionRequestProperties",
-          "description": "Action properties for put request",
+          "$ref": "#/definitions/BookmarkProperties",
+          "description": "Bookmark properties",
           "x-ms-client-flatten": true
         }
       },
       "type": "object"
     },
-    "ActionPropertiesBase": {
-      "description": "Action property bag base.",
+    "BookmarkList": {
+      "description": "List all the bookmarks.",
       "properties": {
-        "logicAppResourceId": {
-          "description": "Logic App Resource Id, providers/Microsoft.Logic/workflows/{WorkflowID}.",
+        "nextLink": {
+          "description": "URL to fetch the next set of cases.",
+          "readOnly": true,
           "type": "string"
+        },
+        "value": {
+          "description": "Array of bookmarks.",
+          "items": {
+            "$ref": "#/definitions/Bookmark"
+          },
+          "type": "array"
         }
       },
       "required": [
-        "logicAppResourceId"
-      ],
-      "type": "object"
+        "value"
+      ]
     },
-    "ActionRequestProperties": {
-      "allOf": [
-        {
-          "$ref": "#/definitions/ActionPropertiesBase"
-        }
-      ],
-      "description": "Action property bag.",
+    "BookmarkProperties": {
+      "description": "Describes bookmark properties",
       "properties": {
-        "triggerUri": {
-          "description": "Logic App Callback URL for this specific workflow.",
+        "created": {
+          "description": "The time the bookmark was created",
+          "format": "date-time",
+          "type": "string"
+        },
+        "createdBy": {
+          "$ref": "#/definitions/UserInfo",
+          "description": "Describes a user that created the bookmark",
+          "type": "object"
+        },
+        "displayName": {
+          "description": "The display name of the bookmark",
+          "type": "string"
+        },
+        "labels": {
+          "description": "List of labels relevant to this bookmark",
+          "items": {
+            "$ref": "#/definitions/Label"
+          },
+          "type": "array"
+        },
+        "notes": {
+          "description": "The notes of the bookmark",
+          "type": "string"
+        },
+        "query": {
+          "description": "The query of the bookmark.",
+          "type": "string"
+        },
+        "queryResult": {
+          "description": "The query result of the bookmark.",
+          "type": "string"
+        },
+        "updated": {
```

Comment:
readonly?

Reply:
same comment - if this is server generated, it must be marked as readonly